src/i18n.ts

```diff
@@ -54,10 +54,12 @@ export const allLanguages = [
 register('en', () => import('./messages/ui.en.json'));
 register('cy', () => import('./messages/ui.cy.json'));
 
-const initialLocale = browserLang({
-  languages: allLanguages.map(l => l.id),
-  fallback: 'en',
-});
+const initialLocale =
+  getLocaleFromQueryString('l') ||
+  browserLang({
+    languages: allLanguages.map(l => l.id),
+    fallback: 'en',
+  });
```

Comment:
Thanks, good point. I've added `getLocaleFromQueryString('l')` into the `initialLocale`. I assumed that the query string language would take precedence over the preferred language in browser, which is what I think [browserLang](https://www.npmjs.com/package/browser-lang) does. I took a second look at the classroom settings.tsx and it seems the default language doesn't take into account the preferred language in the browser: https://github.com/microbit-foundation/classroom-microbit-org/blob/9cae88db315e0f70da9872a9cf5aabd963f807a1/src/settings/settings.tsx#L303C13-L303C14